porcupine/actions.py

```diff
@@ -0,0 +1,140 @@
+from dataclasses import dataclass
+from functools import partial
+from pathlib import Path
+from typing import Callable
+
+from porcupine.tabs import FileTab
+
+action_availability_callback = Callable[[], bool]
+
+
+@dataclass(kw_only=True, slots=True, frozen=True)
+class Action:
```

Comment:
Might be nice to name this something else than `Action`. It will likely need to be special-cased somewhere when looping over all actions, and `for action in get_actions()` + `if isinstance(action, Action)` would look just confusing.

But I'm having trouble coming up with a good name. `SimpleAction`? `NoInputsAction`?

Reply:
I agree with this, what about `RootAction` or `StateAction`, (all?) actions will need some kind of state, the non-path and non-filetab ones would just get it from `state` probably?

Reply:
For now I'd suggest sticking with `get_main_window()` / `get_tab_manager()` instead of injecting. We can consider changing it later, but that would be quite a big change and well beyond the scope of the PR :)

I don't really like either naming suggestion:
- `RootAction` sounds like you can only have one of them, e.g. root file system or root user.
- `StateAction` is confusing without the context you have in mind, because it doesn't have any kind of state or access to the state that other actions lack.

Reply:
BareAction?

Reply:
fixed by 9c330d3